LDMP/api.py

```diff
@@ -27,22 +27,119 @@
 from qgis.PyQt.QtWidgets import QMessageBox
 
 from qgis.utils import iface
+from qgis.core import (
+    QgsAuthMethodConfig,
+    QgsApplication
+)
 
 from LDMP.worker import AbstractWorker, start_worker
 
 from LDMP import log, debug_on
 
 API_URL = 'https://api.trends.earth'
 TIMEOUT = 20
-
+AUTH_CONFIG_NAME = 'Trends.Earth'
 
 class tr_api(object):
     def tr(message):
         return QCoreApplication.translate("tr_api", message)
 
+def init_auth_config(email=None):
+    currentAuthConfig = None
+
+    # check if an auth method with name AUTH_CONFIG_NAME was already set
+    configs = QgsApplication.authManager().availableAuthMethodConfigs()
+    previousAuthExist = False
+    for config in configs.values():
+        if config.name() == AUTH_CONFIG_NAME:
+            currentAuthConfig = config
+            previousAuthExist = True
+            break
+
+    if not previousAuthExist:
+        # not found => craete a new one
+        currentAuthConfig = QgsAuthMethodConfig()
+        currentAuthConfig.setName(AUTH_CONFIG_NAME)
+
+    # reset it's config values to set later new password when received
+    currentAuthConfig.setMethod('Basic')
+    currentAuthConfig.setUri(API_URL + '/auth')
+    currentAuthConfig.setConfig('username', email)
+    currentAuthConfig.setConfig('password', None)
+    currentAuthConfig.setConfig('realm', None)
+
+    if not previousAuthExist:
+        # store a new auth config
+        if not QgsApplication.authManager().storeAuthenticationConfig(currentAuthConfig):
+            iface.messageBar().pushCritical('Trends.Earth', tr_api.tr('Cannot init auth configuration'))
```

Comment:
Can we use a message bar inside plugin UI instead of the iface one.

Reply:
For me ok... may we ask opinion to @ricardogsilva just to verify in what direction to go?
BTW are you referring to message bar inside QgsAuthSelect GUI or another message bar to be added in Setting GUI?

Reply:
I agree with @Samweli that messages related to something that is going on inside some extra dialogue that is currently being used by the user should preferably reported in a message bar of that dialogue - unless that is painful to implement for now.

Reply:
ok, so we should add a general message bar for the DlgSettings ora message bar for each subsection of the DlgSettings or just limit auth messages using the already available messagebar inside QgsAuthSelect ?

Reply:
IMHO a general message bar for DlgSettings dialog would be enough,

Reply:
yes, I agree. Go with your best judgement @luipir

Reply:
ok, doing it

Reply:
@Samweli I'm adding a message bar inside ther DlgSettings not in the plugin docker GUI because settings is activating indipendently by the docked window. Is it ok for you?.
Any sub windows in DlgSettings should have a method to call kind of parent.logMessage
let me know... in the meantime I will proceed adding the message bar to the dialog

Reply:
@ricardogsilva and @Samweli I've added a new singleton messagebar class that attach a message to a dialog when it is shown and reset it to general message bar when the dialog is hided.
In this way any API or Worker will show messages in the current Dialog (when set) instead of to the general one.

Reply:
p.s for sure can be wrote better and probably can be simplified the interface to the singleton messagebar... any suggestion is welcome!

Reply:
> @Samweli I'm adding a message bar inside ther DlgSettings not in the plugin docker GUI because settings is activating indipendently by the docked window. Is it ok for you?

yes @luipir that is okay with me